src/components/GiftRanking/GiftItemCard.tsx

```diff
@@ -1,11 +1,16 @@
 import styled from '@emotion/styled';
-
+import { useNavigate } from 'react-router-dom'; 
 import type { GiftItem } from '@/constants/GiftItem';
 
-
 const GiftItemCard = ({ item }: { item: GiftItem }) => {
+  const navigate = useNavigate(); 
+
+  const handleClick = () => {
+    navigate(`/order/${item.id}`);
```

Comment:
현재 템플릿 리터럴로 다이나믹 경로를 생성하고 있으신데 아무래도 직접 경로 문자열을 만들기 때문에 휴먼에러가 발생할 가능성이 있어요.

React Router의 [generatePath](https://reactrouter.com/6.30.1/utils/generate-path) 유틸을 활용하면 경로 패턴을 상수로 관리하면서 매개변수를 안전하게 주입할 수 있어 편하더라구요.

```suggestion
    import { ORDER_ROUTE } from '...'; // '/order/:id';

    navigate(generatePath(ORDER_ROUTE), {
        id: item.id
    }); // '/order/:id'의 `:id`에 item.id 삽입 
```